### PR TITLE
Fix extraneous blank line on parent ticket description copy

### PR DESCRIPTION
**Pull request by vaevictis**
## Jira ticket
[This way please](https://zendesk.atlassian.net/browse/OFFAPPS-825)

## Description
When the child ticket only contained parent ticket description's copy, an extra white line was added atop it.

Now, the top white line is only added before parent ticket description if the child ticket has a description of its own.

Also fixes an issue when agent is in Markdown mode: the newly created ticket description was stored as escaped `html`.

### Before
white space

### After
no white space

## Risk
Low. Fix existing white line bug in V2 app

### CODE REVIEW

File: src/javascript/app.js [hunk 1]
```diff
@@ -377,27 +377,42 @@ var App = {
...
-    let formDescription = this.formGetOrSet('.description').split(descriptionDelimiter)
-    let returned = formDescription[0]
-
-    if (formDescription.length === 1) {
-      returned += descriptionDelimiter + newTicketDescription
+    let delimiter = `--- ${this.I18n.t('delimiter')} ---${newLine}`
+    let descriptions = this.formGetOrSet('.description').split(delimiter)
+    let returnValue = this.richTextTrim(descriptions[0])
+
+    if (descriptions.length === 1) { // like [''] or ['child description...']: cases 1 & 2
```
Comment (ocke) [inline src/javascript/app.js]: i like doing `(something.length)` or `(!something.length)`, unless you really want to make sure it's exactly `x` amount of elements

Comment (vaevictis) [reply src/javascript/app.js]: In this case, it's exactly `x` amount. I need to know if the array contains exactly 1 element.

File: src/javascript/app.js [hunk 2]
```diff
@@ -377,27 +377,42 @@ var App = {
...
+      if (descriptions[0].length !== 0) { // like ['child description...']: case 2
```
Comment (ocke) [inline src/javascript/app.js]: so then here you can do `if (descriptions[0].length)` :)

File: src/javascript/app.js [hunk 3]
```diff
@@ -377,27 +377,42 @@ var App = {
...
-      returned += descriptionDelimiter + newTicketDescription
+    let delimiter = `--- ${this.I18n.t('delimiter')} ---${newLine}`
+    let descriptions = this.formGetOrSet('.description').split(delimiter)
+    let returnValue = this.richTextTrim(descriptions[0])
+
+    if (descriptions.length === 1) { // like [''] or ['child description...']: cases 1 & 2
+      if (descriptions[0].length) { // like ['child description...']: case 2
+        returnValue += newLine + newLine + delimiter + parentDescription
+      } else { // like ['']: case 1
+        returnValue += delimiter + parentDescription
```
Comment (ocke) [inline src/javascript/app.js]: how do we ever reach the this code?

File: src/javascript/app.js [hunk 4]
```diff
@@ -377,27 +377,42 @@ var App = {
     })
   },
 
-  formatDescription: function (rawTicketDescription, useRichText) {
-    let newTicketDescription
+  // There are four use cases:
```
Comment (ocke) [inline src/javascript/app.js]: love these comments

**Pull request merged**